### PR TITLE
Initiate threading on main file

### LINKED ISSUE #186: Save wind data while flying
### Describe the solution you'd like

be able to save the wind data while flying

### LINKED ISSUE #188: Save EVO TOF sensor data when flying
### Describe the solution you'd like

Be able to save Evo TOF data to check if the sensor work while flying

### LINKED ISSUE #196: Initiate threading from main.py
### Describe the solution you'd like

We need to initiate threading for the barometer from the main file.

### PR DESCRIPTION
**Pull request by Runway27**
## Main Changes

The threading now begins on the main file. 

## Associated Issues

- Closes #196 
- Closes #188 
- Closes #186

## Tests

Has been tested on the quadcopter and all the sensors are collecting the data.

### CODE REVIEW

File: raspberry/main.py [hunk 1]
```diff
@@ -369,48 +374,24 @@ def read_ToF_run_kf_and_LQR():
...
-            print("[LOGGER] saving complete")
-            clear_caches()
-            allow_data_logging[0] = False
-
-            # TODO change this to print the dataframe (all rows and columns)
-            if enable_printing_cache_to_screen[0]:
-                print(f"time: {time_cache} \naltitude: {tof.altitude_cache()} \nTref: {throttle_ref_cache} \nyaw: {yaw_cache} \npitch: {pitch_cache} \nroll:{roll_cache} \nacc: {accelerometer_cache_} \nalti_ref_mts{altitude_reference_cache_mts}")
-
-        allow_data_logging[0] = not switch_a_status[0]
+    EVO_filename = datetime.datetime.now().strftime("EVOSENSOR-LOGS-%d-%m-%y--%H-%M.csv")
```
Comment (jamie-54) [inline raspberry/main.py:377]: Think you should change "EVOSENSOR-LOGS-..." to something like "Evo-ToF-..."

File: raspberry/main.py [hunk 2]
```diff
@@ -369,48 +374,24 @@ def read_ToF_run_kf_and_LQR():
...
+    BAR_filename = datetime.datetime.now().strftime("PressureSensor-%d-%m-%y--%H-%M.csv")
+    ANE_filename = datetime.datetime.now().strftime("Anemometer-%d-%m-%y--%H-%M.csv")
+    processor = AverageFilter()  # You need to define this class based on your requirements
+    with EvoSensor(window_length=3,
```
Comment (jamie-54) [inline raspberry/main.py:381]: Think we should change the layout of this to 

```python
    with (EvoSensor(window_length=3,
                    data_processor=processor,
                    log_file=EVO_filename) as TofSensor, 
          PressureSensor(window_length=100,
                         data_processor=processor,
                         reference_pressure_at_sea_level=102500, 
                         log_file=BAR_filename) as PSensor, 
          Anemometer(window_length=5,
                     data_processor=processor,
                     log_file=ANE_filename) as ASensor):
```

What's your thoughts?

**Comment by jamie-54** [comment]
@Runway27 I've made a view suggestions but overall lgtm

File: raspberry/main.py (hunk 2 above)
Comment (Runway27) [reply raspberry/main.py:381]: Sure

**Comment by jamie-54** [comment]
LGTM
just check it matches PEP8 max line length 79 etc.

Should we merge #201 first then the GNSS can be added to this? what does everyone else think?

**Comment by jamie-54** [comment]
I've Added GNSS to main so think this is ready to merge

**Review by jamie-54 — APPROVED** [review]
LGTM

**Comment by pdavid747** [comment]
looks all good to me

**Comment by Yuanbwcx** [comment]
Is it correct that the sensor data has not been specified yet?

**Comment by jamie-54** [comment]
> Is it correct that the sensor data has not been specified yet?

sorry not sure what you mean?

**Comment by Yuanbwcx** [comment]
> > Is it correct that the sensor data has not been specified yet?
> 
> sorry not sure what you mean?

Get it now, my bad

**Pull request merged**